openapi.yaml

```diff
@@ -4711,16 +4711,35 @@ components:
       example: null
       description: |-
         The namespace the `process_id` is valid for.
+
+        The following options are pre-defined by the openEO API, but additional
+        namespaces may be introduced by back-ends or in a future version of the API.
         
-        By default, the namespace is `null`, which includes all pre-defined processes listed at `GET /processes`.
-        The namespace `user` refers to all user-defined processes listed as `GET /process_graphs`.
-        Additional namespaces may be introduced by back-ends or in a future version of the API.
+        * `null` (default): Checks both user-defined and pre-defined processes,
+           but prefers user-defined processes if both are available.
+           This allows users to replace missing pre-defined processes with
+           user-defined processes for portability, e.g. common processes from
+           [processes.openeo.org](https://processes.openeo.org) that have a process graph
+           included.
+        * `backend`: Uses exclusively the pre-defined processes listed at `GET /processes`.
+        * `user`: Uses exclusively the user-defined processes listed at `GET /process_graphs`.
+
+        Note on collisions: The only way a collision between user-defined processes and
+        pre-defined processes can occur is when back-ends introduce a new process with an
+        identifier already used by users. User-defined processes with an identifier used by
+        a pre-defined process MUST be rejected. To ensure reproducibility, back-ends SHOULD
+        persist the namespace that is prioritized at the time the process is submitted.
+        If a collision occurs after a has back-end introduced a new pre-defined process
+        (i.e. a user-defined process is prioritized over a pre-defined process),
+        a log entry should be added so that the user is informed about the availability of the
+        new process.
```

Comment:
> a log entry should be added
Would this be available under /jobs ? Assuming th job works, I doubt most users will ever look in there, so they might not be aware that a new process exists.

Reply:
I agree that a log entry is not a very strong signal

Reply:
Yes, in the job or service logs. It will likely not work or give unexpected results, I guess. Otherwise the processes would be at least quite similar anyway (parameter names and schemas must match).

Reply:
Now clients should warn, but back-ends can should still also warn, I think.